### ISSUE TITLE
publish WebJob as a netcoreapp application

### ISSUE THREAD

**Issue by SergeyRudakov** [Azure/azure-webjobs-sdk#745]
Hello,
I  would like to publish a WebJob from .NET Core console app.
It seems that Microsoft.Azure.WebJobs doesn't support Microsoft.NETCore.App 1.0 and there is no "Publish as Azure WebJob..." support from VS 2015.
Let me know, what would be the proper way to deploy .Net core app as a WebJob?

Here is my frameworks section:

```
 "frameworks": {

    "netcoreapp1.0": {
      "imports": [
        "dotnet5.6",
        "dnxcore50",
        "portable-net45+win8"
      ]
    }
  }

```

Thanks,

**Comment by SergeyRudakov** [comment]
BTW: I was able to upload WebJob to the site using Kudu.
Job was published as self contained app with 

`  "runtimes": {
    "win7-x64": { }
    },`

App runs successfully on local environment but failed to run in Azure with these errors:
`
[06/08/2016 22:21:34 > f0962e: ERR ] Failed to load the dll from [D:\local\Temp\jobs\continuous\LogProcessor\ui2ijlq2.iyq\coreclr.dll], HRESULT: 0x8007007E
[06/08/2016 22:21:34 > f0962e: ERR ] 
[06/08/2016 22:21:34 > f0962e: ERR ] Failed to bind to CoreCLR at [D:\local\Temp\jobs\continuous\LogProcessor\ui2ijlq2.iyq]
`
Here is environment info:

OS version: Microsoft Windows NT 6.2.9200.0
64 bit system: True
64 bit process: True

**Comment by SergeyRudakov** [comment]
I was able to run the app somehow but only through manual Kudu copy.

However it seems standard WebJob features are not supported because **Microsoft.Azure.WebJobs** is not compatible with netcoreapp framework. Is there any plans to this port library?

**Comment by christopheranderson** [comment]
dotnet core support is a goal, but we have no plans for it yet. We'll consider this after WebJobs 2.0.0 is rtm'd.

**Comment by PaulMouchet** [comment]
Is there any sort of ETA on when WebJobs might be able to support .NetCore?  In the meantime though, how would you suggest implementing a webjob-like process using .NetCore?

**Comment by davidebbo** [comment]
To clarify, Azure WebJobs that use .NET Core work fine. It's specifically the use of the WebJobs **SDK** that is not yet supported.

**Comment by PaulMouchet** [comment]
Yes.  My question was so poorly worded I'm not really sure what the heck I was thinking.  Any ETA on when the SDK will be supported?  Is there a REST API similar to what is available for the Service Bus?

**Comment by christopheranderson** [comment]
We don't have an ETA on this. Not this year, likely.

**Comment by DreamingDevs** [comment]
@christopheranderson Can we know what is the alternative solution?

**Comment by jonsagara** [comment]
Hi,

It has been a while since the last update. Do you have any sort of ETA or road map for .NET Core support?

Thanks!

**Comment by paulbatum** [comment]
Investigation work for Azure Functions on .NET Core is now in progress and of course since Functions is built on WebJobs, it is included in the effort. You can track the latest here:
https://github.com/Azure/Azure-Functions/issues/98

**Comment by tbertenshaw** [comment]
since SDK V2 has now been released (https://github.com/Azure/azure-webjobs-sdk/releases/tag/v2.0.0), does the SDK support dotnetcore webjobs? And are there any examples/or visualstudio project templates?

**Comment by ggirard07** [comment]
Any update on this as .Net Core support is now here with `v3.0.0`?
Trying to publish a webjob ported to netcoreapp2.1 with webjobs SDK 3.0.0 but can't figure out the proper to get it published right...

**Comment by paulbatum** [comment]
If you grab the latest tools as described in the [prerequisites section of this doc](https://docs.microsoft.com/en-us/azure/azure-functions/functions-develop-vs#prerequisites) (it applies for both webjobs and functions) then you should be able to just rightclick publish in VS. The extra ".Publish" NuGet is no longer required.

**Comment by davidebbo** [comment]
Link to same question on StackOverflow: https://stackoverflow.com/questions/52612500/how-to-publish-webjobs-targeting-netcoreapp2-1-with-sdk-webjobs-sdk-3-0-to-azure

**Comment by ggirard07** [comment]
@davidebbo I am the author of the StackOverflow question
@paulbatum actually it helped a lot in some sort! by following your instructions, I created another publish profile which solved my issue. I don't know why but first profile I created did not update my `.csproj` to add the `;Microsoft.NET.Sdk.Publish` reference to `Project` tag. By following your instruction, I selected the `Publish` option in the wizard instead of `Create profile` which I think is what updated my `.csproj` this time.

With the added reference to `;Microsoft.NET.Sdk.Publish` publish is working perfectly and Kudu pick ups my webjo and start it in `Continuous` mode as configured

**Comment by davidebbo** [comment]
@ggirard07 Yes, I know the question was from you. But it's important to keep things linked (in both direction) so that all the context can easily be found from either place.

@vijayrkn, you probably understand this better than anyone else, so it would be great of you could shed some light on where things stand with the ability to publish Core WebJobs from VS. Thanks!

**Comment by vijayrkn** [comment]
As mentioned in the StackOverflow answer, there is support in VS for publishing .NET Core webjobs. It just works like any other publish operation.

Right click on a .NET core app -> Publish 0> Select App Service & Publish. This should push the WebJobs to Azure.

**Comment by vijayrkn** [comment]
@ggirard07 - The SDK attribute is added at the time of first publish. So it should work for both Continuous and triggered. I just tried both 'Publish immediately' and 'create profile' and both scenarios worked when I clicked publish from VS.

After creating the profile from VS, are you publishing from a different location outside VS?

**Comment by ggirard07** [comment]
@vijayrkn Indeed, retested locally and everything is updated properly now... I can't understand what happened the first time. I had build errors in my output related to VSTS package feed returning 401 unauthorized despite no packages coming from that feed. Don't know whether this could have had some impacts.

I have a bunch of remaining webjobs to update, will notify you if it happens again. Sorry for the inconvenience.

**Comment by vijayrkn** [comment]
@ggirard07 - No problem. Glad to know that it is working for you. Let us know if you run into any issues.

**Comment by ggirard07** [comment]
@vijayrkn actually I just ran into an issue related to the fact webjobs publishing now rely on a publish profile. Locally everything is working fine. How do you handle CI on platforms like VSTS?

Publish profile are not added to the source code and should not in that case as the same build artifact is going to be deployed to multiple environments anyway. Currently my VSTS build basically generates an empty project package (contains only the base folder `obj\Release\netcoreapp2.1\PubTmp\Out` which remains empty).

Is there any ways to provide msbuild parameters now defined in the publish profile directly in the csproj? Aim is to start the build command on the solution to build multiple webjobs at the same time.

**Comment by vijayrkn** [comment]
>How do you handle CI on platforms like VSTS?

You can add any property in the pubxml to csproj (pubxml is simply an msbuild file).  Also, you can check in the pubxml file. It should not contain any authentication info.

The decision to not combine the publishing for WebApps and WebJobs (for Sdk style projects) was deliberate. Since the publishing process for the WebJob project is just msbuild based, it should be just similar to how the Web App is published. 

https://github.com/aspnet/websdk/issues/296 - I have created an item here to make this experience better. Hope this helps!
